### DISCUSSION TITLE
Unable to build .aab with 33.1.1.929, JAVA_HOME is not defined error

### DISCUSSION THREAD

**Discussion by kasparv** [airsdk/Adobe-Runtime-Support#2116] [Q&A] [answered]
Hi! 

I'm having trouble getting the latest SDK (33.1.1.929) to build .aab while 33.1.1.633 works fine with the same setup - 

Here's my build command:

`$FLEXSDK/bin/adt -package -target aab -storetype PKCS12 -keystore tool/temp_keystore.p12 -storepass keystore_password Mobile-$VERSION.aab AppMobile-app-generated.xml -extdir ane -C  ${PRODUCTNAME}Mobile.swf -C v4${PRODUCT}_src icon -C v4${PRODUCT}_src package -C ane com.distriqt.NetworkInfo.ane -C ane com.distriqt.Core.ane -C ane com.distriqt.Share.ane -C ane androidx.core.ane -platformsdk ../../SDKs/android-sdk`

The error I'm getting is this - 

> NOTE: Multiple Android build-tools folders exist, choosing the latest version (29.0.2) for packaging
> WARNING: could not find appropriate Java Home location for v11+ JDK
> JAVA_HOME is not defined: this must point to a valid JDK installation for Android App Bundle creation

Terminal `echo $JAVA_HOME` returns `/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home`

I've tried setting JAVA_HOME in adt.cfg to that and also to `JAVA_HOME=/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home` without anything changing, same with setting `AndroidPlatformSDK` also.

So what has changed since 33.1.1.633 ? Do I need specific versions of Android SDK, Build Tools, Gradle etc?

Thanks in advace!

**Comment by ajwfrost** [comment]
Yes, we had to update the Gradle plug-in for the later Android features which meant other updates were needed, including a change to the Java runtime required...
```
WARNING: could not find appropriate Java Home location for v11+ JDK
```
So basically it's looking for version 11 (or higher), JDK. If you have Android Studio installed, this should include the JDK? (and the ADT tool automatically looks for it if it can't find anything better...)

If you want to change JAVA_HOME just for ADT, you can edit a file ~/.airsdk/adt.cfg and add it there e.g.
```
JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home
```

thanks

**Comment by kasparv** [comment, marked as answer]
Like I wrote above, setting `JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home` in adt.cfg did not change anything.
However I got it to work by uninstalling adoptopenjdk with `brew remove homebrew/cask-versions/adoptopenjdk8` and reinstalling with `brew install java`.

**Reply by ajwfrost** [reply]
Oh yes, sorry I hadn't spotted that bit. Curious, as I would have thought that this should be the Java 11 SDK. Are you able to check e.g. what do you get if you run
```
/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home/bin/java -version
```
(Maybe there's something up with the path/settings or similar..)

Anyway -> glad you managed to get it working, I assume now if you run `java -version` you get something a bit later than 8..

thanks

**Reply by kasparv** [reply]
`/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home/bin/java -version` 
returns 
> openjdk version "1.8.0_242-release"
> OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
> OpenJDK 64-Bit Server VM (build 25.242-b3-6915495, mixed mode)

and `java -version` 
now returns 

> openjdk version "18.0.2.1" 2022-08-18
> OpenJDK Runtime Environment Homebrew (build 18.0.2.1+0)
> OpenJDK 64-Bit Server VM Homebrew (build 18.0.2.1+0, mixed mode, sharing)

Thanks for checking in on this!

**Reply by ajwfrost** [reply]
Oh okay! So your Android Studio was I guess a bit older .. but the new Java version means that the latest Android tools should work, the only thing I know that won't work is the ADT classes for code-signing internally (but you're on a mac, so it will default to using the native "codesign" tool for IPAs and it will use the Android "apksigner" for APKs...)

thanks :-)
